File: like-service/src/main/java/com/backend/like/controller/PostLikeController.java
package com.backend.like.controller;

import com.backend.like.dto.PostLikeDto;
import com.backend.like.service.PostLikeService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/like")
@RequiredArgsConstructor
public class PostLikeController {

  private final PostLikeService postLikeService;

  @PostMapping("like-post")
  @ResponseStatus(HttpStatus.OK)
  public String likePost(@Valid @RequestBody PostLikeDto postLikeDto) {
    return postLikeService.toggleLike(postLikeDto);
  }
}

File: like-service/src/main/kotlin/com/backend/like/controller/PostLikeController.kt
package com.backend.like.controller

import com.backend.like.dto.PostLikeDto
import com.backend.like.service.PostLikeService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/like")
class PostLikeController(private val postLikeService: PostLikeService) {

    @PostMapping("like-post")
    @ResponseStatus(HttpStatus.OK)
    fun likePost(@Valid @RequestBody postLikeDto: PostLikeDto): String {
        return postLikeService.toggleLike(postLikeDto)
    }
}